### PR TITLE
workflows: windows: Use chocolatey's OpenSSL installer for linking OpenSSL

### LINKED ISSUE #5703: fluent-bit 1.9.6 for Windows and lack of OpenSSL dlls (again?)
## Bug Report

**Describe the bug**
image
image

**To Reproduce**
Just try to exec the binary with or without any parameters

OS: Windows 11 Pro and the latest (1.9.6) version of fluent-bit downloaded from official Download page

### PR DESCRIPTION
**Pull request by cosmo0920**
Signed-off-by: Hiroshi Hatake <hatake@calyptia.com>

On GitHub Actions, we need to refer the equivalent OpenSSL libraries than AppVeyor ones.
In AppVeyor, it seems that it installs slproweb's OpenSSL library installers to provide OpenSSL libraries.
We need to use them on GitHub Actions' workflow to solve OpenSSL linking issue on the artifacts.

### With chocolatey's OpenSSL installer (32bit)

image

### With chocolatey's OpenSSL installer (64bit)

image

---

There is no reported error for dll linkage.

Closes #5703
 
----
Enter `[N/A]` in the box, if an item is not applicable to your change.

**Testing**
Before we can approve your change; please submit the following in a comment:
- [N/A] Example configuration file for the change
- [N/A] Debug log output from testing the change

- [N/A] Attached [Valgrind](https://valgrind.org/docs/manual/quick-start.html) output that shows no leaks or memory corruption was found

If this is a change to packaging of containers or native binaries then please confirm it works for all targets.
- [N/A] Attached [local packaging test](./packaging/local-build-all.sh) output showing all targets (including any new ones) build.

**Documentation**

- [N/A] Documentation required for this feature

**Backporting**

- [ ] Backport to latest stable release.

----

Fluent Bit is licensed under Apache 2.0, by submitting this pull request I understand that this code will be released under the terms of that license.

### CODE REVIEW

File: .github/workflows/call-build-windows.yaml [hunk 1]
```diff
@@ -45,14 +45,17 @@ jobs:
...
             arch: x86
-            vcpkg_triplet: x86-windows-static
-            openssl_root_dir_opt: -DOPENSSL_ROOT_DIR=C:\vcpkg\packages\openssl_x86-windows-static
+            win32openssl: https://slproweb.com/download/Win32OpenSSL-1_1_1q.msi
+            win32openssl_dir: "C:\\Program\ Files\\OpenSSL-Win32"
           - name: "Windows 64bit"
             arch: x64
-            vcpkg_triplet: x64-windows-static
-            openssl_root_dir_opt: ""
+            win32openssl: https://slproweb.com/download/Win64OpenSSL-1_1_1q.msi
```
Comment (patrick-stephens) [inline .github/workflows/call-build-windows.yaml:52]: we probably should just name this `openssl_dir` or something rather than `win32` in the 64-bit case...

File: .github/workflows/call-build-windows.yaml [hunk 2]
```diff
@@ -45,14 +45,17 @@ jobs:
...
+            win32openssl_dir: "C:\\Program\ Files\\OpenSSL-Win64"
     permissions:
       contents: read
+    # Default environment variables can be overridden below
+    env:
+        PATH: c:/Program Files/Git/cmd;c:/Windows/system32;C:/Windows/System32/WindowsPowerShell/v1.0;$ENV:WIX/bin;C:/Program Files/CMake/bin
```
Comment (patrick-stephens) [inline .github/workflows/call-build-windows.yaml:58]: What's the reason for this change?

**Review by patrick-stephens — COMMENTED** [review]
Some minor issues with the changes, need to review the licensing of slproweb offering to make sure it is acceptable too.

File: .github/workflows/call-build-windows.yaml (hunk 2 above)
Comment (cosmo0920) [reply .github/workflows/call-build-windows.yaml:58]: This is because another openssl dlls are detected on C:\Strawberry\ or something like that. This detected dlls are harmful for linking.

Comment (patrick-stephens) [reply .github/workflows/call-build-windows.yaml:58]: Ah ok, can you add a comment then for it? Otherwise it'll likely get deleted next time someone tries to clean it up :+1:

File: .github/workflows/call-build-windows.yaml (hunk 1 above)
Comment (cosmo0920) [reply .github/workflows/call-build-windows.yaml]: Yup. I used openssl_dir for this variable.

File: .github/workflows/call-build-windows.yaml [hunk 3]
```diff
@@ -45,14 +45,17 @@ jobs:
...
+            chocolatey_opt: --x86
           - name: "Windows 64bit"
             arch: x64
-            vcpkg_triplet: x64-windows-static
-            openssl_root_dir_opt: ""
+            openssl_dir: C:\Program Files\OpenSSL-Win64
+            chocolatey_opt: --x64
     permissions:
       contents: read
+    # Default environment variables can be overridden below. To prevent library pollution.
```
Comment (patrick-stephens) [inline .github/workflows/call-build-windows.yaml:56]:
```suggestion
    # Default environment variables can be overridden below. To prevent library pollution - without this other random libraries may be found on the path leading to failures.
```

**Review by patrick-stephens — DISMISSED** [review]

File: .github/workflows/call-build-windows.yaml (hunk 3 above)
Comment (cosmo0920) [reply .github/workflows/call-build-windows.yaml]: Ah, great!

**Review by patrick-stephens — APPROVED** [review]

**Comment by patrick-stephens** [comment]
@cosmo0920 can you just confirm you've checked the contents of the built binaries now include the relevant dependencies?

**Comment by cosmo0920** [comment]
This is my run link: https://github.com/fluent/fluent-bit/actions/runs/2732639226
